Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

xichengliudui opened the pull request:
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

**Which issue(s) this PR fixes**:

Ref ##72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

xichengliudui commented on hack/jenkins/test-dockerized.sh:
```diff
@@ -21,7 +21,12 @@ set -o xtrace
+        return 0
+    else
+        sleep "${i}"
+    fi
```
Why did it change and what did it fix:
```
In ./hack/jenkins/test-dockerized.sh line 24:
    "$@" && return 0 || sleep "${i}"
         ^-- SC2015: Note that A && B || C is not if-then-else. C may run when A is true.

For more information:
  https://www.shellcheck.net/wiki/SC2015 -- Note that A && B || C is not if-t...
```
Example: 
**Problematic code:**
```
[[ $dryrun ]] && echo "Would delete file" || rm file
```
**Correct code:**
```
if [[ $dryrun ]]
then
  echo "Would delete file"
else
  rm file
fi
```

xichengliudui commented on hack/jenkins/upload-to-gcs.sh:
```diff
@@ -70,13 +70,13 @@ if [[ -n "${remote_git_repo}" ]]; then
-  : ${JENKINS_GCS_LOGS_INDIRECT:="gs://kubernetes-jenkins/pr-logs/directory/${JOB_NAME}"}
+  : "${JENKINS_GCS_LOGS_PATH:="gs://kubernetes-jenkins/pr-logs/pull/${GCS_SUBDIR}${ghprbPullId:-unknown}"}"
+  : "${JENKINS_GCS_LATEST_PATH:="gs://kubernetes-jenkins/pr-logs/directory"}"
+  : "${JENKINS_GCS_LOGS_INDIRECT:="gs://kubernetes-jenkins/pr-logs/directory/${JOB_NAME}"}"
```
Why did it change and what did it fix:
```
In ./hack/jenkins/upload-to-gcs.sh line 73:
  : ${JENKINS_GCS_LOGS_PATH:="gs://kubernetes-jenkins/pr-logs/pull/${GCS_SUBDIR}${ghprbPullId:-unknown}"}
    ^-- SC2223: This default assignment may cause DoS due to globbing. Quote it.

For more information:
  https://www.shellcheck.net/wiki/SC2223
```

Example:
**Problematic code**:
```
: ${COLUMNS:=80}
```
**Correct code**:
```
: "${COLUMNS:=80}"
```

xichengliudui commented on hack/jenkins/upload-to-gcs.sh:
```diff
@@ -116,20 +116,22 @@ function find_version() {
     cat version
   elif [[ -e "hack/lib/version.sh" ]]; then
     export KUBE_ROOT="."
+    # shellcheck disable=SC1091
```
Why did it change and what did it fix:
```
In ./hack/jenkins/upload-to-gcs.sh line 119:
    source "hack/lib/version.sh"
           ^-------------------^ SC1091: Not following: hack/lib/version.sh was not specified as input (see shellcheck -x).

For more information:
  https://www.shellcheck.net/wiki/SC1091
```

Example:
**Problematic code**:
```
source somefile
```
**Correct code**: 
In case you have access to the file, e.g. if it is included in your source code repository:
```
# shellcheck source=somefile
source somefile
```
In case you do not have access to the file:
```
# shellcheck disable=SC1091
source somefile
```

BenTheElder replied:
this seems like something we should fix in the linter instead of in the source files

xichengliudui replied:
but shellcheck raises a question.

BenTheElder replied:
I filed https://github.com/kubernetes/kubernetes/pull/75977 to prevent this warning. I confirmed that with https://github.com/kubernetes/kubernetes/pull/75977 we do not need to add this line.

BenTheElder replied:
#75977 is in now. We can remove this change.

xichengliudui commented on hack/jenkins/upload-to-gcs.sh:
```diff
@@ -116,20 +116,22 @@ function find_version() {
 function print_started() {
-  local metadata_keys=$(compgen -e | grep ^BUILD_METADATA_)
+  metadata_keys=$(compgen -e | grep ^BUILD_METADATA_)
+  local metadata_keys
```
Why did it change and what did it fix:
```
In ./hack/jenkins/upload-to-gcs.sh line 132:
  local metadata_keys=$(compgen -e | grep ^BUILD_METADATA_)
        ^-----------^ SC2155: Declare and assign separately to avoid masking return values.
```

BenTheElder replied:
local has to come before assignment

xichengliudui replied:
I have some questions about it. According to SC2155, I should have no problem to modify it like this.

BenTheElder replied:
no, it needs to be first:
https://github.com/koalaman/shellcheck/wiki/SC2155#correct-code-1

the reason is `local` creates a variable with a reduced scope, the scope cannot be reduced retroactively by merely re-declaring it.

BenTheElder replied:
this was fixed, thanks! resolving

xichengliudui commented on hack/jenkins/upload-to-gcs.sh:
```diff
@@ -153,7 +155,7 @@ function print_started() {
 if [[ -n "${TEST_STARTED_JSON:-}" ]]; then
   version=$(find_version)
-  cat <(print_started) | jq .
+  < <(print_started) jq .
```
Why did it change and what did it fix:
```
In ./hack/jenkins/upload-to-gcs.sh line 156:
  cat <(print_started) | jq .
      ^--------------^ SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.

For more information:
  https://github.com/koalaman/shellcheck/wiki/SC2002
```
Example:
**Problematic code**:
```
cat file | tr ' ' _ | nl
cat file | while IFS= read -r i; do echo "${i%?}"; done
```
**Correct code**:
```
< file tr ' ' _ | nl  
while IFS= read -r i; do echo "${i%?}"; done < file
```

BenTheElder replied:
I don't think this change works correctly. `jq . <(print_started)` seems like the correct change.

xichengliudui replied:
thanks.

xichengliudui commented:
/assign @cblecker @BenTheElder

xichengliudui commented:
/test pull-kubernetes-e2e-gce

soltysh committed:
Fix create secret generic paths examples

Add RUNTIME_REQUEST_TIMEOUT to local-up-cluster.sh

The changes to local-cluster-up.sh from #75662 is causing the
CRI-O CI to break since it picks up the --runtime-request-timeout
flag as part of the container-runtime-endpoint flag.
Adding a new env so that we can set the runtime-request-timeout
value for our CI.

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

Extract Method for kubeadm

Signed-off-by: PingWang <wang.ping5@zte.com.cn>

update pull request

update pull request

update pull request

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

xichengliudui commented:
@BenTheElder 
I have merged #75977 and updated pr, is that ok?

BenTheElder commented:
Please do not include merge commits in your PR 😅 

You can undo this by checking `git reflog` for the ref prior to merging and then doing a `git reset` to that ref (carefully!). Please use `git fetch` like `git fetch upstream` with `git rebase` like `git rebase upstream/master` instead of merging.

There are more docs for the workflow in https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md

The reason for this is when we merge your PR we do not want merge commits that were not to the target branch to be in the git history, it can be confusing for people looking through the history when triaging failures.

BenTheElder commented:
For what it's worth: You actually don't need to do anything with my PR, it doesn't have conflicting changes with this one and the CI will always perform a local merge of your PR into the target branch before testing, which will include any already merged changes.

xichengliudui committed:
Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

Fix shellcheck failures in test-dockerized.sh and upload-to-gcs.sh

xichengliudui commented:
Has been updated.

yagonobre commented:
/remove-area kubeadm
/remove-sig cluster-lifecycle

xichengliudui commented:
/test pull-kubernetes-integration

BenTheElder commented:
i think the commit is still not right 😅 
<img width="960" alt="Screen Shot 2019-04-02 at 10 29 52 PM" src="https://user-images.githubusercontent.com/917931/55455059-fbd4e680-5596-11e9-9188-0d9ebe0fc5d4.png">

appears to be combined with the author and description of https://github.com/kubernetes/kubernetes/commit/38fe68658d06fb7ab765d3c4013856181bc4f0e7 ?

BenTheElder commented:
generally the changes look good though, thank you 🙏

xichengliudui commented:
/remove-area kubectl
